alg/teca_model_segmentation.py

```diff
@@ -0,0 +1,196 @@
+try:
+    from mpi4py import MPI
+    comm = MPI.COMM_WORLD
+    rank = comm.Get_rank()
+except:
+    rank = 0
+import teca_py
+import numpy as np
+import torch
+import torch.nn.functional as F
+
+def teca_load_state_dict(state_dict_file):
```

Comment:
we need to pass the comm in to this function. The passed comm instance should come from a call to teca_algorithm::get_communicator. This gives us runtime control over which ranks work collectively.